### PR TITLE
docs(guide): converting /tutorials/projection.md to rst

### PR DESCRIPTION
**Pull request by daprahamian**
Fixes NODE-2209

## Description

**What changed?**

Added a big warning about trying to project with the second parameter of `find`.

**Are there any files to ignore?**

### CODE REVIEW

File: docs/guide/tutorials/projections.txt [hunk 1]
```diff
@@ -1,3 +1,89 @@
...
+documents. To limit the amount of data that MongoDB sends to
+applications, you can include a projection document in the query
+operation.
+
+Projection Document
+-------------------
+
+The projection document limits the fields to return for all
+matching documents. The projection document can specify the
+inclusion of fields or the exclusion of field and has the
```
Comment (mbroadst) [inline docs/guide/tutorials/projections.txt]: *fields

File: docs/guide/tutorials/projections.txt [hunk 2]
```diff
@@ -1,3 +1,89 @@
...
+   // Connection URL
+   const url = 'mongodb://localhost:27017';
+   
+   // Database Name
+   const dbName = 'myproject';
+   
+   // Create a new MongoClient
+   const client = new MongoClient(url);
+   
+   // BEGIN foobar
```
Comment (mbroadst) [inline docs/guide/tutorials/projections.txt]: what is this line?

File: docs/guide/tutorials/projections.txt [hunk 3]
```diff
@@ -1,3 +1,89 @@
...
+   // Use connect method to connect to the Server
+   client.connect().then(async function() {
+     console.log("Connected successfully to server");
+   
+     const collection = client.db(dbName).collection('restaurants');
+
+     try {
+       const results = collection
```
Comment (mbroadst) [inline docs/guide/tutorials/projections.txt]: you need to `await` this line

File: docs/guide/tutorials/projections.txt [hunk 4]
```diff
@@ -1,3 +1,89 @@
...
+         .find({ cuisine : 'Brazilian' })
+         .project({ name : 1, cuisine : 1 })
+         .toArray();
+       
+       console.log("Found the following records");
+       console.log(docs)
```
Comment (mbroadst) [inline docs/guide/tutorials/projections.txt]: missing semicolon

File: docs/guide/tutorials/projections.txt [hunk 5]
```diff
@@ -1,3 +1,89 @@
...
+       await client.close();
+     }
+   });
+
+To return ``name`` and ``cuisine`` but exclude all other fields,
+including ``_id``, use the following projection document:
+
+.. code-block:: js
+
+   { 'name' : 1, 'cuisine' : 1, '_id': 0 }
```
Comment (mbroadst) [inline docs/guide/tutorials/projections.txt]: do you want to wrap the properties with quotation marks? The code snippets don't do this.

File: docs/guide/tutorials/projections.txt [hunk 6]
```diff
@@ -1,3 +1,88 @@
...
+   const client = new MongoClient(url);
+
+   // Use connect method to connect to the Server
+   client.connect().then(async function() {
+     console.log("Connected successfully to server");
+   
+     const collection = client.db(dbName).collection('restaurants');
+
+     try {
+       const results = await collection
```
Comment (mongomoe) [inline docs/guide/tutorials/projections.txt]: variable `results` isn't used. Did you mean to print this out? If not can you remove the variable and just `await collection...`

File: docs/guide/tutorials/projections.txt [hunk 7]
```diff
@@ -1,3 +1,88 @@
...
+   const url = 'mongodb://localhost:27017';
+   
+   // Database Name
+   const dbName = 'myproject';
+   
+   // Create a new MongoClient
+   const client = new MongoClient(url);
+
+   // Use connect method to connect to the Server
+   client.connect().then(async function() {
```
Comment (mongomoe) [inline docs/guide/tutorials/projections.txt]: ` client.connect().then(async function() {` should be es6 arrow fn : `client.connect().then(async () => {`

File: docs/guide/tutorials/projections.txt [hunk 8]
```diff
@@ -1,3 +1,88 @@
...
+     const collection = client.db(dbName).collection('restaurants');
+
+     try {
+       const results = await collection
+         .find({ cuisine : 'Brazilian' })
+         .project({ name : 1, cuisine : 1 })
+         .toArray();
+       
+       console.log("Found the following records");
+       console.log(docs);
```
Comment (mongomoe) [inline docs/guide/tutorials/projections.txt]: `docs` isn't defined. I think you meant to log `results`

File: docs/guide/tutorials/projections.txt (hunk 7 above)
Comment (daprahamian) [reply docs/guide/tutorials/projections.txt]: Holding off on this change until #2182

**Pull request merged**